core/src/main/java/org/testcontainers/containers/wait/HostPortWaitStrategy.java

```diff
@@ -23,87 +22,35 @@
 @Slf4j
 public class HostPortWaitStrategy extends GenericContainer.AbstractWaitStrategy {
 
-    private static final String SUCCESS_MARKER = "TESTCONTAINERS_SUCCESS";
     @Override
     protected void waitUntilReady() {
-        final Integer port = getLivenessCheckPort();
-        if (null == port) {
-            log.debug("Liveness check port of {} is empty. Not waiting.", container.getContainerName());
+        final Set<Integer> externalLivenessCheckPorts = getLivenessCheckPorts();
+        if (externalLivenessCheckPorts.isEmpty()) {
+            log.debug("Liveness check ports of {} is empty. Not waiting.", container.getContainerName());
             return;
         }
 
-        Callable<Boolean> checkStrategy;
+        List<Integer> exposedPorts = container.getExposedPorts();
 
-        if (shouldCheckWithCommand()) {
-            List<Integer> exposedPorts = container.getExposedPorts();
+        final Set<Integer> internalPorts = exposedPorts.stream()
```

Comment:
Again I would prefer to see a private method, `getInternalPorts()`. Sorry for being so picky about private methods, that's a coding style I prefer, you don't have to follow if you aren't a fan of it, just think it makes it easier to get a high level understanding of what's happening, especially for people more unfamiliar with the codebase 😁

Reply:
No, that's a good point so thanks for bringing it up (all three!)

Reply:
Will fix :)

Reply:
If only we had Extensions...!